### PR TITLE
feat: re-add QD to tekton-ci

### PR DESCRIPTION
**Pull request by psturc**
[KONFLUX-2244](https://issues.redhat.com/browse/KONFLUX-2244)

Tested together with changes in https://github.com/redhat-appstudio/infra-deployments/pull/3370

### CODE REVIEW

File: frontend/Dockerfile [hunk 1]
```diff
@@ -16,6 +16,13 @@ RUN chmod +x entrypoint.sh
 RUN yarn build
 RUN chgrp -R 0 /app && \
     chmod -R g=u /app
+RUN rm -rf ./node_modules
+
+FROM node:18-alpine
+WORKDIR /app
+
+COPY --from=builder /app .
+RUN yarn global add sirv-cli
```
Comment (psturc) [inline frontend/Dockerfile:25]: this reduces the size of the image significantly
Screenshot 2024-03-27 at 21 08 29

File: frontend/entrypoint.sh [hunk 2]
```diff
@@ -6,6 +6,9 @@ DEX_ISSUER_REPLACE=$(grep DEX_ISSUER .env | cut -d '=' -f 2)
 FRONTEND_REDIRECT_URI_REPLACE=$(grep FRONTEND_REDIRECT_URI .env | cut -d '=' -f 2)
 DEX_APPLICATION_ID_REPLACE=$(grep DEX_APPLICATION_ID .env | cut -d '=' -f 2)
 
+cp -r dist package.json .env /tmp/
+cd /tmp
+
```
Comment (psturc) [inline frontend/entrypoint.sh:11]: This is required due to [readOnlyRootFilesystem requirement](https://github.com/redhat-appstudio/infra-deployments/pull/3370), which is needed for https://github.com/redhat-appstudio/infra-deployments/pull/3370 to pass the kube-linter check.

I introduced [a new volume mount](https://github.com/redhat-appstudio/quality-dashboard/pull/312/files#diff-54d683ca0912dc7d95eeb59691820b0ee573002c53c30cf1b4f5fec35a062337R28-R30) for mounting `/tmp` where the frontend dist files are moved to and updating by the shell script.

**Review by flacatus — APPROVED** [review]

**Pull request merged**